### PR TITLE
User can now configure multiple instances of a tool to exist

### PR DESCRIPTION
**Pull request by bpagon13**

### CODE REVIEW

**Comment by cglewis** [comment]
@bpagon13 let's do some work on the codeclimate issues and codacy issues.  also some minimal tests would be appreciated since it's a such a large PR.

**Comment by bpagon13** [comment]
Got it, will work on that.

**Comment by cglewis** [comment]
@bpagon13 I was going to try out this PR, but when I went to configure I noticed that `instances` is removed from the configure with your patch somehow.

screen shot 2017-08-25 at 9 28 05 am

screen shot 2017-08-25 at 9 28 15 am

**Comment by bpagon13** [comment]
I actually purposefully did that because I don't want that information to get overwritten by users. I want them to specifically use the add/remove instances option instead.

**Comment by cglewis** [comment]
where is the add/remove instances option you mention?

**Comment by bpagon13** [comment]
Okay yeah I'll talk to Jeff about adding that. In the meantime, it's in configure tools. Once you're in there if you click ^B you will be able to configure tools by the number of instances they have. I couldn't add another section to the core tools and plugin tools menu options because of that weird npyscreen bug.

**Comment by cglewis** [comment]
ah, I see.  So by default then, running core tools without configuring should start 2 instances of `rq_worker`. I just tried that on your patch and only 7 of the core tools started and only 1 instance of `rq_worker`.

**Comment by bpagon13** [comment]
Weird, I was not having that issue. I'll double check.

**Comment by bpagon13** [comment]
I just removed all vent images, containers, and the .vent folder to start from scratch and installed and ran everything properly.

**Comment by bpagon13** [comment]
Just merged your latest commits (6 commits) from the past few hours to see if it still works with those.

**Comment by bpagon13** [comment]
Still not able to reproduce that issue.

**Comment by cglewis** [comment]
ok, i'll give it another whirl with a clean clone and see what i get.

File: docs/source/core_and_plugins.rst [hunk 1]
```diff
@@ -99,7 +99,15 @@ Stop and remove the chosen tools' containers.
...
 Edit a tool's vent.template file found in the tool's respective folder.
-Read more about :ref:`venttemplate-label`
+Read more about :ref:`venttemplate-label`.
+
+Another option you have is to configure the number of instances that are
+running of that tool. You can toggle between configuring the vent.template
+for a tool or the number of instances running for that tool by hitting
+CTRL-B. Configuring the number of instances running for a tool would be
+useful for scaling. An example would be if you wanted multiple workers
+to be handle all that tasks that you are processing or want different
```
Comment (cglewis) [inline docs/source/core_and_plugins.rst]: `to be` -> `to be able to`

**Comment by cglewis** [comment]
ok, it looks happier now, not sure what happened before.  a couple things - it seems kind of confusing to have it in the manifest as 2 entries, which also causes things like the status to look incorrect:

screen shot 2017-08-25 at 1 56 12 pm

I need to do some more testing on this, but user experience and expectation is a little goofy.  maybe @Jeff-Wang93 or @lilchurro can chime in with suggestions to make this a bit more intuitive?

**Comment by bpagon13** [comment]
Okay I could look at restructuring it to not have multiple instances in the manifest. I had that in there because the user can then set custom run time settings for any instances that they want. The manifest is where we normally store that run time customization information.

**Comment by bpagon13** [comment]
Also, I could get rid of the Ctrl-B and allow users to directly enter in how many instances they want. I'm thinking this would probably be more intuitive to an end user.

**Comment by cglewis** [comment]
> Also, I could get rid of the Ctrl-B and allow users to directly enter in how many instances they want. I'm thinking this would probably be more intuitive to an end user.

Yeah, I think that would probably make more sense.

> Okay I could look at restructuring it to not have multiple instances in the manifest. I had that in there because the user can then set custom run time settings for any instances that they want. The manifest is where we normally store that run time customization information.

Interesting, I could go either way on that, didn't think of that.  Let's see if @Jeff-Wang93 or @lilchurro have any thoughts on which way to go.

**Comment by Jeff-Wang93** [comment]
More options to a user is (to me) always welcomed so I feel like 2 entries in the manifest is fine. Will think about how to display the information cleanly.

**Comment by cglewis** [comment]
for now let's keep it as separate entries in the manifest like you have, but think about better way to accurately display status of built/running tools.  also update the ctrl-b change discussed above.

**Pull request merged**